app/models/subscription.rb

```diff
@@ -0,0 +1,35 @@
+class Subscription < ApplicationRecord
+  
+  belongs_to :owner, class_name: :User
+  belongs_to :plan
+  has_many :subscription_users
+  has_many :users, through: :subscription_users
+
+  validates :start_date, presence: true
+  validate :ensure_user_is_not_an_active_subscriber
```

Comment:
add a validation that expiry date cannot be less than start_date

Reply:
Added.